fin_control/src/fin_control.cpp

```diff
@@ -261,107 +260,58 @@ void FinControl::handleSetAngles(const fin_control::SetAngles::ConstPtr &msg)
-  }
-
-  if (currentLoggingEnabled)
+  // check for max angle the fins can mechanically handle
+  for (uint8_t i = 0; i < 4; i++)
   {
-    int32_t cdata1, cdata2, cdata3, cdata4;
-    cdata1 = cdata2 = cdata3 = cdata4 = 0;
+    angle_plus_offset = ctrlFinScaleFactor * (msg->fin_angle_in_radians[i] + ctrlFinOffset);
 
-    if (!myWorkBench.itemRead(1, "Present_Current", &cdata1, &log))
-    {
-      ROS_ERROR("Could Read F1 Current %s", log);
-    }
-    else if (!myWorkBench.itemRead(2, "Present_Current", &cdata2, &log))
+    if (!(myWorkBench.addBulkWriteParam(i + 1, "Goal_Position",
+                                        myWorkBench.convertRadian2Value(i + 1, angle_plus_offset),
+                                        &log)))
     {
-      ROS_ERROR("Could Read F2 Current %s", log);
-    }
-    else if (!myWorkBench.itemRead(3, "Present_Current", &cdata3, &log))
-    {
-      ROS_ERROR("Could Read F3 Current %s", log);
+      ROS_ERROR_STREAM("Could not add bulk write param " << i + 1 << " " << log);
+      return;
     }
-    else if (!myWorkBench.itemRead(4, "Present_Current", &cdata4, &log))
+
+    if (!myWorkBench.bulkWrite(&log))
     {
-      ROS_ERROR("Could Read F4 Current %s", log);
+      ROS_ERROR("Could not bulk write %s", log);
     }
-    else
+
+    if (currentLoggingEnabled)
     {
-      ROS_INFO("Current for fin 1-4 %f,%f,%f,%f", myWorkBench.convertValue2Current((int16_t)cdata1),
-               myWorkBench.convertValue2Current((int16_t)cdata2),
-               myWorkBench.convertValue2Current((int16_t)cdata3),
-               myWorkBench.convertValue2Current((int16_t)cdata4));
-    }
-  }
-}
+      int32_t cdata1, cdata2, cdata3, cdata4;
+      cdata1 = cdata2 = cdata3 = cdata4 = 0;
 
-void FinControl::handleSetAngle(const fin_control::SetAngle::ConstPtr &msg)
-{
-// check for max angle the fins can mechanically handle
-  if (fabs(msg->angle_in_radians) > maxCtrlFinAngle)
-  {
-    ROS_ERROR("Angle set is out of range: %f rad", msg->angle_in_radians);
-    return;
+      if (!myWorkBench.itemRead(1, "Present_Current", &cdata1, &log))
+      {
+        ROS_ERROR("Could Read F1 Current %s", log);
+      }
+      else if (!myWorkBench.itemRead(2, "Present_Current", &cdata2, &log))
+      {
+        ROS_ERROR("Could Read F2 Current %s", log);
+      }
+      else if (!myWorkBench.itemRead(3, "Present_Current", &cdata3, &log))
+      {
+        ROS_ERROR("Could Read F3 Current %s", log);
+      }
+      else if (!myWorkBench.itemRead(4, "Present_Current", &cdata4, &log))
+      {
+        ROS_ERROR("Could Read F4 Current %s", log);
+      }
+      else
+      {
+        ROS_INFO("Current for fin 1-4 %f,%f,%f,%f",
+                 myWorkBench.convertValue2Current((int16_t)cdata1),
+                 myWorkBench.convertValue2Current((int16_t)cdata2),
+                 myWorkBench.convertValue2Current((int16_t)cdata3),
+                 myWorkBench.convertValue2Current((int16_t)cdata4));
+      }
```

Comment:
@patapufate consider reading current as you loop over IDs.

Reply:
fixed in 772413b